easybuild/tools/toolchain/toolchain.py

```diff
@@ -243,13 +245,19 @@ def set_minimal_build_env(self):
 
         # this is only relevant when using a system toolchain,
         # for proper toolchains these variables will get set via the call to set_variables()
-        env_vars = {
-            'CC': 'gcc',
-            'CXX': 'g++',
-            'F77': 'gfortran',
-            'F90': 'gfortran',
-            'FC': 'gfortran',
+        known_sys_compilers = {
+            'gcc': ['CC'],
+            'g++': ['CXX'],
+            'gfortran': ['F77', 'F90', 'FC'],
         }
+
+        env_vars = {}
+        for compiler, flags in known_sys_compilers.items():
+            # distutils.spawn.find_executable() only returns first hit from $PATH
+            if find_executable(compiler):
```

Comment:
Why not use the `which` function we have (in the `filetools`) module?

It think it also makes sense to add some logging here, to print the path to the command that was found for example.

If we don't set the variable for `gfortran` (I don't think it makes sense), then we can also print a warning when `gcc` or `g++` is not found?

I'll make this changes myself in my PR https://github.com/easybuilders/easybuild-framework/pull/3399

Reply:
I had no idea that EB already had a `which` function :laughing: